lib/ts/recipe/openid/types.ts

```diff
@@ -0,0 +1,86 @@
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import OverrideableBuilder from "supertokens-js-override";
+import { BaseRequest, BaseResponse } from "../../framework";
+import NormalisedURLDomain from "../../normalisedURLDomain";
+import NormalisedURLPath from "../../normalisedURLPath";
+import { RecipeInterface as JWTRecipeInterface, APIInterface as JWTAPIInterface } from "../jwt/types";
+
+export type TypeInput = {
+    issuer: string;
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+        jwtFeature?: {
+            functions?: (
+                originalImplementation: JWTRecipeInterface,
+                builder?: OverrideableBuilder<JWTRecipeInterface>
+            ) => JWTRecipeInterface;
+            apis?: (
+                originalImplementation: JWTAPIInterface,
+                builder?: OverrideableBuilder<JWTAPIInterface>
+            ) => JWTAPIInterface;
+        };
+    };
+};
+
+export type TypeNormalisedInput = {
+    issuerDomain: NormalisedURLDomain;
+    issuerPath: NormalisedURLPath;
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+        jwtFeature?: {
+            functions?: (
+                originalImplementation: JWTRecipeInterface,
+                builder?: OverrideableBuilder<JWTRecipeInterface>
+            ) => JWTRecipeInterface;
+            apis?: (
+                originalImplementation: JWTAPIInterface,
+                builder?: OverrideableBuilder<JWTAPIInterface>
+            ) => JWTAPIInterface;
+        };
+    };
+};
+
+export type APIOptions = {
+    recipeImplementation: RecipeInterface;
+    config: TypeNormalisedInput;
+    recipeId: string;
+    req: BaseRequest;
+    res: BaseResponse;
+};
+
+export type APIInterface = {
+    getOpenIdDiscoveryConfigurationGET:
+        | undefined
+        | ((input: { options: APIOptions }) => Promise<DiscoveryConfiguration>);
+};
+
+export type DiscoveryConfiguration = {
+    status: "OK";
+    issuer: string;
+    jwks_uri: string;
+};
+
+export type RecipeInterface = {
+    getDiscoveryConfiguration(): Promise<DiscoveryConfiguration>;
```

Comment:
Change name to `getOpenIdDiscoveryConfiguration`

Reply:
Done